Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp

```diff
@@ -857,33 +874,50 @@ void NetworkStorageManager::setItem(IPC::Connection& connection, StorageAreaIden
     ASSERT(!RunLoop::isMain());
 
     bool hasQuotaError = false;
-    if (auto storageArea = m_storageAreaRegistry->getStorageArea(identifier)) {
-        auto result = storageArea->setItem(connection.uniqueID(), implIdentifier, String { key }, WTFMove(value), WTFMove(urlString));
-        if (!result)
-            hasQuotaError = (result.error() == StorageError::QuotaExceeded);
-    }
+    auto storageArea = m_storageAreaRegistry->getStorageArea(identifier);
+    if (!storageArea)
+        return completionHandler(hasQuotaError);
 
+    auto result = storageArea->setItem(connection.uniqueID(), implIdentifier, String { key }, WTFMove(value), WTFMove(urlString));
+    if (!result)
+        hasQuotaError = (result.error() == StorageError::QuotaExceeded);
     completionHandler(hasQuotaError);
+
+    writeOriginToFileIf(storageArea->origin(), [&]() {
+        return storageArea->type() == StorageAreaBase::Type::SQLite && !storageArea->isEmpty();
+    });
 }
 
 void NetworkStorageManager::removeItem(IPC::Connection& connection, StorageAreaIdentifier identifier, StorageAreaImplIdentifier implIdentifier, String&& key, String&& urlString, CompletionHandler<void()>&& completionHandler)
 {
     ASSERT(!RunLoop::isMain());
 
-    if (auto storageArea = m_storageAreaRegistry->getStorageArea(identifier))
-        storageArea->removeItem(connection.uniqueID(), implIdentifier, WTFMove(key), WTFMove(urlString));
-
+    auto storageArea = m_storageAreaRegistry->getStorageArea(identifier);
+    if (!storageArea)
+        return completionHandler();
+    
+    storageArea->removeItem(connection.uniqueID(), implIdentifier, WTFMove(key), WTFMove(urlString));
     completionHandler();
+
+    writeOriginToFileIf(storageArea->origin(), [&]() {
+        return storageArea->type() == StorageAreaBase::Type::SQLite && !storageArea->isEmpty();
```

Comment:
This is repeated so many times, this should be factored in its own function to avoid repetition, e.g.:
```
if (shouldWriteOriginToFile(storageArea))
    writeOriginToFile(storageArea->origin());
```

Reply:
Sure